six/modules/c++/cphd/include/cphd/Metadata.h

```diff
@@ -71,6 +71,8 @@ struct Metadata : MetadataBase
     size_t getCompressedSignalSize(size_t channel) const override;
     bool isCompressed() const override;
 
+    std::string getUri() const;
```

Comment:
I don't think the uri stuff is necessary here. The only usage I can see is in `CPHDXMLControl::toXMLImpl`. But the function it's calling wants a version, not a uri.

Reply:
So I implemented the CPHDXMLParser constructor such that it takes in a URI, as opposed to the sicd parsers which take in a version to build the URI (See `six:ComplexXMLParser` constructor).
Sicd is able to do that because they expect a fixed format for the URI in all the schemas (i.e. urn::SICD:x.x.x). So it just plugs in the appropriate version in the string and pass the schema validation check.
But because the CPHD1.0.1 uri is different, I think the only way to do write out the same uri would be to just store the URI somewhere and use it again.

Reply:
Ok, that makes sense. Just update the relevant parameter names in `CPHDXMLControl` to reflect that.